src/snowflake/snowpark/dataframe.py

```diff
@@ -433,25 +433,34 @@ def stat(self) -> DataFrameStatFunctions:
         return self._stat
 
     @df_action_telemetry
-    def collect(self) -> List["Row"]:
+    def collect(
+        self, *, _statement_params: Optional[Dict[str, Any]] = None
+    ) -> List["Row"]:
         """Executes the query representing this DataFrame and returns the result as a
         list of :class:`Row` objects.
+
+        Args:
+            _statement_params: Extra information that should be sent to Snowflake with query.
         """
-        return self._internal_collect_with_tag()
+        return self._internal_collect_with_tag(_statement_params=_statement_params)
 
-    def _internal_collect_with_tag(self) -> List["Row"]:
+    def _internal_collect_with_tag(
+        self, *, _statement_params: Optional[Dict[str, Any]] = None
+    ) -> List["Row"]:
         # When executing a DataFrame in any method of snowpark (either public or private),
         # we should always call this method instead of collect(), to make sure the
         # query tag is set properly.
         return self._session._conn.execute(
             self._plan,
-            _statement_params={"QUERY_TAG": create_statement_query_tag(3)}
-            if not self._session.query_tag
-            else None,
+            _statement_params=create_or_update_statement_params_with_query_tag(
+                _statement_params, self._session.query_tag, 3
```

Comment:
Can we make this 3 as a constant in utils.py?

Reply:
Also, might worth add a comment about this magic number.

Reply:
good suggestion, updated to be a constant with notes (as well as the magic number 2)